### PR TITLE
[AIRFLOW-XXXX] Add architecture section to k8sexec docs

### PR DESCRIPTION
**Pull request by dimberman**
---
Issue link: `Document only change, no JIRA issue`

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by mik-laj** [comment]
Can you add some labels to the diagram, please? The diagram uses the Airflow logo to describe many different components.  It is not clear what the role of each logo is. Lines appear in the diagram, but they are not described. I think that it is also worth numbering the next steps in the diagram, which allows for easier tracking of the message. If possible, it is worth using generally accepted conventions, e.g. sequence diagram. This makes the drawings simpler to understand and easier to draw.  You could use online editor e.g. https://www.planttext.com/  
Example: https://github.com/apache/airflow/pull/6961/files 
Old preview: https://user-images.githubusercontent.com/12058428/71550253-ebd62000-29cb-11ea-827f-d6d66cf25256.png
What do you think about Did you have any reason to draw this diagram in a graphics program?  I am afraid of such diagram, because then they are difficult to edit.


I have one more comment. Can you maintain uniform naming between text and diagram also, please? You use the term "the backend DB" in the text, but it does not appear in the image, but instead is "PostgresSQL". 

I also prepared preview of your documentation:
Preview:  https://acceptable-goat.surge.sh/executor/kubernetes.html

File: docs/executor/kubernetes.rst [hunk 1]
```diff
@@ -34,3 +34,71 @@ The volumes are optional and depend on your configuration. There are two volumes
   - By storing logs onto a persistent disk, the files are accessible by workers and the webserver. If you don't configure this, the logs will be lost after the worker pods shuts down
 
   - Another option is to use S3/GCS/etc to store logs
+
+KubernetesExecutor Architecture
+################################
+
+The KubernetesExecutor runs as a single pod that only requires access to the Kubernetes API. There are no external brokers or consistent workers needed.
```
Comment (ashb) [inline docs/executor/kubernetes.rst]: It is possible to run KubeExecutor from outside of a cluster too. Perhaps we can ignore that here though.

Also the KubeExecutor doesn't run as a pod, but runs inside the scheduler.

No external brokers, but still need the (existing) metadata DB, which can't be sqlite (let's be explicit.)

Comment (dimberman) [reply docs/executor/kubernetes.rst]: These are all good points. Will fix now.

File: docs/executor/kubernetes.rst [hunk 2]
```diff
@@ -34,3 +34,71 @@ The volumes are optional and depend on your configuration. There are two volumes
...
+
+===========================
+Handling Worker Pod Crashes
+===========================
+
+When dealing with distributed systems, we need a system that assumes that any component can crash at any moment for reasons ranging from OOM errors to node upgrades.
+
+In the case where a worker dies before it can report its status to the backend DB, the executor can use a Kubernetes watcher thread to discover the failed pod.
+
+.. image:: ../img/k8s-watcher-2.jpeg
```
Comment (kaxil) [inline docs/executor/kubernetes.rst]: We don't have this image !

**Review by kaxil — APPROVED** [review]

**Pull request merged**